src/alcotest-engine/core.ml

```diff
@@ -393,16 +393,24 @@ struct
     |> List.sort compare_path
     |> Fmt.(list ~sep:(const string "\n") (pp_info t) stdout)
 
-  let validate_name name =
-    let pattern = "^[a-zA-Z0-9 _-]+$" in
-    let re = Re.(compile @@ Pcre.re pattern) in
-    if not (Re.execp re name) then
-      let msg =
-        Fmt.strf "%a %S is not a valid test label (must match %s)." red "Error:"
-          name pattern
-      in
-      Error msg
-    else Ok ()
+  let get_codepoint buf u =
+    Buffer.add_string buf (Printf.sprintf "U+%04X" (Uchar.to_int u))
+
+  let normalize_name (name : string) =
+    let buf = Buffer.create (String.length name * 2) in
+    let get_normalized_char _ _ u =
+      match u with
+      | `Uchar u ->
+          if Uchar.is_char u then
+            match Uchar.to_char u with
+            | ('A' .. 'Z' | 'a' .. 'z' | '0' .. '9' | '_' | '-' | ' ') as c ->
+                Buffer.add_char buf c
+            | _ -> get_codepoint buf u
+          else get_codepoint buf u
+      | `Malformed _ -> Uutf.Buffer.add_utf_8 buf Uutf.u_rep
+    in
+    Uutf.String.fold_utf_8 get_normalized_char () name;
+    Ok (Buffer.contents buf)
```

Comment:
```suggestion
    Buffer.contents buf
```

Looks like we don't need to return an option here anymore, since there is no failure case :heart_eyes:

I suspect that means we can also get rid of the `Result.map` and `List.lift_result` calls below.